fido/src/main/java/com/yubico/yubikit/fido/ctap/BioEnrollment.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2024 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.yubikit.fido.ctap;
+
+import com.yubico.yubikit.core.application.CommandException;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Implements Bio enrollment commands.
+ *
+ * @see <a href="https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-errata-20220621.html#authenticatorBioEnrollment">authenticatorBioEnrollment</a>
+ */
+public class BioEnrollment {
+    protected static final int RESULT_MODALITY = 0x01;
+    protected static final int MODALITY_FINGERPRINT = 0x01;
+
+    protected final Ctap2Session ctap;
+    protected final int modality;
+
+    public BioEnrollment(Ctap2Session ctap, int modality) throws IOException, CommandException {
+        if (!isSupported(ctap.getCachedInfo())) {
+            throw new IllegalStateException("Bio enrollment not supported");
+        }
+
+        this.ctap = ctap;
+        this.modality = getModality();
```

Comment:
Not a problem right now, but: calling instance methods in the constructor can be a bit dangerous because there's no language-enforced guarantee that the object is fully constructed yet. So this could break if at some point the `this.ctap = ctap;` line gets deleted or moved to after the `getModality()` call (in other words, the correctness of the method depends on circumstances outside the method definition). Perhaps better to make `getModality()` a static function (I think it could also be private?) and take the `ctap` as a parameter instead?

Reply:
great suggestion!